File: resources/scop_1.5.1/python/examples/rtt_client.py
#!/usr/bin/python
# rtt_client.cpp - DMI - 8-11-03

# Usage: rtt_client [ <iterations> ]   (default is 1000)

import scop, sys, time

REPEATS = 1000

if len(sys.argv) == 2:
	REPEATS = int(sys.argv[1])
sock = scop.scop_open("localhost", "rtt_client")

tv_start = time.time()
for i in range(REPEATS):
	reply = scop.scop_rpc(sock, "server", "test")
	if reply != "tteesstt":
		print "Message error."
		sys.exit()
tv_end = time.time()

diff = tv_end - tv_start
print str(REPEATS) + " round trips in " + str(diff) + " seconds."
print "Time each = " + str(diff * 1000000 / REPEATS) + " us."
print str(REPEATS / diff) + " iterations per second."
scop.scop_close(sock)

File: resources/scop_1.5.1/examples/rtt_client.cpp
// rtt_client.cpp - DMI - 11-9-02

/* Usage: rtt_client [ <iterations> ]   (default is 1000) */

#include <scop.h>

#include <sys/time.h>

int REPEATS = 1000;

int main(int argc, char **argv)
{
	int sock;
	char *reply;
	struct timeval tv_start, tv_end;
	long us;
	
	if(argc == 2)
		REPEATS = atoi(argv[1]);
	
	sock = scop_open("localhost", "rtt_client");
	gettimeofday(&tv_start, NULL);
	for(int i = 0; i < REPEATS; i++)
	{
		reply = scop_rpc(sock, "server", "test");
		if(strcmp(reply, "tteesstt"))
		{
			printf("Message error.\n");
			exit(0);
		}
		delete[] reply;
	}
	gettimeofday(&tv_end, NULL);
	us = (tv_end.tv_sec - tv_start.tv_sec) * 1000000;
	us += tv_end.tv_usec - tv_start.tv_usec;
	printf("%d round trips in %d us; time each = %d us; approx %d per second.\n",
			REPEATS, us, us / REPEATS, 1000000 / (us / REPEATS));
	
	close(sock);
	return 0;
}
